pkg/i2gw/provider.go

```diff
@@ -99,8 +101,90 @@ type GatewayResources struct {
 	UDPRoutes  map[types.NamespacedName]gatewayv1alpha2.UDPRoute
 
 	ReferenceGrants map[types.NamespacedName]gatewayv1beta1.ReferenceGrant
+
+	GatewayExtensions map[types.NamespacedName]unstructured.Unstructured
+}
+
+// IR contains all Gateway-API objects, and Intermediate Representation(IR) to
```

Comment:
curious what's going to happen to the `GatewayResources` struct

Reply:
shouldnt the `IR` replace it ?

Reply:
IR and Gateway Resources are for different translation purpuse. What I have in mind is:

A pool of resources
------- > Ingress + Ingress extension (By input provider reader)
------- > IR, or Gateway/HTTPRoute context(By input provider converter)
------->  Gateway + Gateway extension(By output provider extension converter)

Reply:
Let's say we are doing translation of enableHTTPRedirect feature, we will have:

type GceGatewayIR struct {
   EnableHTTPRedirect bool
}
type GceHTTPRouteIR struct {
   EnableHTTPRedirect bool
}

And during translation, what would happen is:
Ingress+Ingress extension
--->IR.GceGatewayIR.EnableHTTPRedirect=true, IR.GceHTTPRouteIR.EnableHTTPRedirect=true
--->Gateway with additional HTTPRoute

Gateway extensions will be translated in a similar manner.